### ISSUE TITLE
Error 413 after successful login

### ISSUE THREAD

**Issue by BenPortner** [aws-samples/cloudfront-authorization-at-edge#117]
Dear developers,

thank you for the great login functionality! I successfully deployed the solution using one of my existing CloudFront distributions. However, I am struggling to do the same for another distribution. After logging in, I receive an error 413 because the URL string is longer than the maximum permitted by CloudFront. For some reason, the value of the `state` parameter is longer than 9000 characters. I don't understand why this happens. The settings of both CloudFront distributions are identical, except they point to different S3 buckets and use different lambda functions (all of them were created using your template). Both S3 buckets contain a different SPA. All lambda functions point to the same user pool. CNAMES were configured for both distributions and in the cognito app settings. Do you have any idea what could be going wrong?

All the best.
Ben

**Comment by ottokruse** [comment]
Hi Ben,
Mmm ... strange.

The state is created like this:

```
      state:
        // Encode the state variable as base64 to avoid a bug in Cognito hosted UI when using multiple identity providers
        // Cognito decodes the URL, causing a malformed link due to the JSON string, and results in an empty 400 response from Cognito.
        urlSafe.stringify(
          Buffer.from(
            JSON.stringify({ nonce: state.nonce, requestedUri })
          ).toString("base64")
        ),
```
https://github.com/aws-samples/cloudfront-authorization-at-edge/blob/7cd699a15e1989c0f888ff72450ef03ddd43d9e8/src/lambda-edge/check-auth/index.ts#L119

Which must mean I think that the requestedUri is super long? (Or we have a bug and a ridiculous long nonce is calculated)

Can you base64 decode the state param and have a look what is in there then?

**Comment by BenPortner** [comment]
Dear @ottokruse,

thanks for your quick reply. I solved the problem meanwhile by creating a new CloudFront instance. I already deleted the old one, so I cannot check the state param anymore. I am certain that the nonce was fine, though. My guess is that there was a problem with the previous lambda function, which I used to enable basic HTTP authentification (based on [this](https://hackernoon.com/serverless-password-protecting-a-static-website-in-an-aws-s3-bucket-bfaaa01b8666) solution). I had removed the corresponding behavior but my guess is that it was still active somehow. Either way, it seems like the problem is not with your code :)

Sorry for the extra work. Hope this is still helpful for other people experiencing the similar problems. Keep up the great work!

Ben
